model/project_ref.go

```diff
@@ -230,48 +223,61 @@ func (p *GitHubDynamicTokenPermissionGroup) Intersection(other GitHubDynamicToke
 
 	// Iterate through all of their fields.
 	for i := 0; i < perms1.NumField(); i++ {
-		perm1, ok := perms1.Field(i).Interface().(*string)
+		perm1Ptr, ok := perms1.Field(i).Interface().(*string)
 		// This currently should not get triggered, but if GitHub
 		// introduces a field that isn't a pointer to a string-
 		// this will stop a wide spread panic.
 		if !ok {
 			continue
 		}
-		perm2, ok := perms2.Field(i).Interface().(*string)
+		perm2Ptr, ok := perms2.Field(i).Interface().(*string)
 		if !ok {
 			continue
 		}
 
-		// If either are nil, that counts as the most restrictive
-		// permission (no permission).
-		if perm1 == nil || perm2 == nil {
-			continue
+		mostRestrictivePermission := "invalid"
```

Comment:
removed the comment suggesting using "", because I realized that can be a valid "most restrictive permission".